shelley/chain-and-ledger/executable-spec/test/Rules/ClassifyTraces.hs

```diff
@@ -79,3 +71,10 @@ lenRatio :: ([a] -> [b]) -> [a] -> Double
 lenRatio f xs
   = ratioInt (length (f xs))
              (length xs)
+
+onlyValidLedgerSignalsAreGenerated :: Property
+onlyValidLedgerSignalsAreGenerated =
+  withTests 200 $
+    onlyValidSignalsAreGeneratedForTrace traceGen
+  where
+    traceGen = traceOfLengthWithInitState @LEDGER 100 mkGenesisLedgerState
```

Comment:
I should have asked this before, but can't the genesis state be generated from the environment?

Reply:
In this case the genesis state (utxo0) is independent of the environment